indexing-hadoop/src/main/java/org/apache/druid/indexer/JobHelper.java

```diff
@@ -386,29 +386,13 @@ public static void writeJobIdToFile(String hadoopJobIdFileName, String hadoopJob
     }
   }
 
-  public static boolean runSingleJob(Jobby job, HadoopDruidIndexerConfig config)
+  public static boolean runSingleJob(Jobby job)
   {
     boolean succeeded = job.run();
-
-    if (!config.getSchema().getTuningConfig().isLeaveIntermediate()) {
```

Comment:
Hm, would removing the cleanup here prevent HadoopDruidDetermineConfigurationJob from doing any necessary cleanup?

Reply:
Good catch! Fixed. Also found another place where we were maybe not doing necessary cleanup.

Reply:
Changed it so that the segment descriptor for a segment is not overwritten if it already exists, as we discussed.